mycroft/client/enclosure/display_manager.py

```diff
@@ -0,0 +1,164 @@
+from threading import Thread, Timer
+from mycroft.messagebus.client.ws import WebsocketClient
+from mycroft.configuration import ConfigurationManager
+from mycroft.util import get_ipc_directory
+import json
+import os
+from logging import getLogger
+
+__author__ = 'connorpenrod', 'michaelnguyen'
+
+
+LOG = getLogger("Display Manager (mycroft.client.enclosure)")
+managerIPCDir = os.path.join(get_ipc_directory(), "managers")
+
+
+def write_data(dictionary):
+    """Writes the parama as JSON to the
+        IPC dir (/tmp/mycroft/ipc/managers)
+        args:
+            dict: dictionary
+    """
+
+    # change read/write permissions based on if file exists or not
+    path = os.path.join(managerIPCDir, "disp_info")
+    permission = "r+" if os.path.isfile(path) else "w+"
+
+    if permission == "w+" and os.path.isdir(managerIPCDir) is False:
+        os.makedirs(managerIPCDir)
+
+    try:
+        with open(path, permission) as dispFile:
+
+            # check if file is empty
+            if os.stat(str(dispFile.name)).st_size != 0:
+                data = json.load(dispFile)
+
+            else:
+                data = {}
+                LOG.info("Display Manager is creating " + dispFile.name)
+
+            for key in dictionary:
+                data[key] = dictionary[key]
+
+            dispFile.seek(0)
+            dispFile.write(json.dumps(data))
+            dispFile.truncate()
+
+    except Exception as e:
+        LOG.error(e)
+        LOG.error("Error found in display manager file, deleting...")
+        os.remove(path)
+        write_data(dictionary)
+
+
+def read_data():
+    """ Reads the file in (/tmp/mycroft/ipc/managers/disp_info)
+        and returns the the data as python dict
+    """
+
+    path = os.path.join(managerIPCDir, "disp_info")
+    permission = "r" if os.path.isfile(path) else "w+"
+
+    if permission == "w+" and os.path.isdir(managerIPCDir) is False:
+        os.makedirs(managerIPCDir)
+
+    data = {}
+    try:
+        with open(path, permission) as dispFile:
+
+            if os.stat(str(dispFile.name)).st_size != 0:
+                data = json.load(dispFile)
+
+    except Exception as e:
+        LOG.error(e)
+        os.remove(path)
+        read_data()
+
+    return data
+
+
+def set_active(skill_name):
+    """ Sets skill name as active in the display Manager
+        args:
+            string: skill_name
+    """
+    write_data({"active_skill": skill_name})
+    LOG.info("Setting active skill to " + skill_name)
+
+
+def get_active():
+    """ Get active skill in the display manager
+    """
+    data = read_data()
+    active_skill = ""
+
+    if "active_skill" in data:
+        active_skill = data["active_skill"]
+
+    return active_skill
+
+
+def remove_active():
+    """ Remove the active skill in the skill manager
+    """
+    LOG.error("Removing active skill...")
+    write_data({"active_skill": ""})
+
+
+def run():
+    """ TODO: document
```

Comment:
Perhaps rename to `initiate_ws` or `initiate_display_manager_ws` (like it was imported as) and make the doc string the comment written near the function ("initiates the web sockets on display manager").